src/MiniCover/Commands/Options/SolutionDirOption.cs

```diff
@@ -0,0 +1,45 @@
+﻿using Microsoft.Extensions.CommandLineUtils;
+using System;
+using System.IO;
+
+namespace MiniCover.Commands.Options
+{
+    public class SolutionDirOption : IMiniCoverOption<DirectoryInfo>
+    {
+        private const string DefaultValue = "./";
+        private bool _invalidated;
+        private CommandOption _option;
+        private DirectoryInfo _value;
+        public string Description => $"Solution directory [default: {DefaultValue}]";
+        public string OptionTemplate => "--solutiondir";
+        public CommandOptionType Type => CommandOptionType.SingleValue;
+
+        public DirectoryInfo Value
+        {
+            get
+            {
+                if (_invalidated) return _value;
+                throw new MemberAccessException("Option should be invalidate before Value access");
```

Comment:
I think it would be much simpler if we have just:
```
public DirectoryInfo Value { get; private set; }
```
Or maybe move the logic that ensures `Validate` was called to a baseclass. So we don't need to do that on every option.